runtime/backtrace_byt.c

```diff
@@ -304,16 +304,17 @@ code_t caml_next_frame_pointer(value* stack_high, value ** sp,
 {
   while (*sp < stack_high) {
     value *spv = (*sp)++;
-    code_t *p;
-    if (Is_long(*spv)) continue;
-    p = (code_t*) spv;
-    if((code_t*)&Trap_pc(stack_high + *trap_spoff) == p) {
+    code_t pc;
+    if (Is_block(*spv)) continue;
+    if(&Trap_pc(stack_high + *trap_spoff) == spv) {
       *trap_spoff = Trap_link(stack_high + *trap_spoff);
       continue;
     }
 
-    if (find_debug_info(*p) != NULL)
-      return *p;
+    pc = Ptr_val(*spv);
+
+    if (find_debug_info(pc) != NULL)
+      return pc;
```

Comment:
Before, this function would call `find_debug_info` on all stack slots that are blocks. Now we call `find_debug_info` on all stack slots that are long/immediate. Is there not a risk of performance regression here, if there are more immediates than blocks on the stack in typical programs? (Of course those functions are less critical than the GC, but I'm still asking out of curiosity.)

Reply:
Yeah, that's true. Perhaps we could work around it by adding a fast-path to find_debug_info for small values (e.g. <4k) - those aren't valid pointers, and I imagine most immediates are small numbers.

Reply:
That's an excellent idea; but if you are thinking of the zero page, is it also present on ocaml-freestanding systems?